refinedstorage-fabric/src/main/java/com/refinedmods/refinedstorage/fabric/mixin/AbstractContainerMenuMixin.java

```diff
@@ -0,0 +1,60 @@
+package com.refinedmods.refinedstorage.fabric.mixin;
+
+import com.refinedmods.refinedstorage.common.api.RefinedStorageApi;
+import com.refinedmods.refinedstorage.common.api.storage.StorageContainerItemHelper;
+
+import net.minecraft.core.NonNullList;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.inventory.ClickType;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(AbstractContainerMenu.class)
+public abstract class AbstractContainerMenuMixin {
+    @Shadow
+    @Final
+    public NonNullList<Slot> slots;
+
+    @Shadow
+    public abstract ItemStack getCarried();
+
+    @Shadow
+    public abstract void setCarried(ItemStack stack);
+
+    @Inject(method = "doClick", at = @At("HEAD"), cancellable = true)
+    public void doClick(final int slotId,
+                        final int button,
+                        final ClickType clickType,
+                        final Player player,
+                        final CallbackInfo ci) {
+        if (clickType != ClickType.CLONE
+            || !player.hasInfiniteMaterials()
+            || !this.getCarried().isEmpty()
+            || slotId < 0) {
+            return;
+        }
+
+        final Slot slot = this.slots.get(slotId);
+        if (!slot.hasItem()) {
+            return;
+        }
+
+        final ItemStack stack = slot.getItem();
+        final StorageContainerItemHelper helper = RefinedStorageApi.INSTANCE.getStorageContainerItemHelper();
+
+        if (helper.getId(stack).isEmpty()) {
+            return;
+        }
+
+        helper.removeId(stack);
+        this.setCarried(stack.copyWithCount(stack.getMaxStackSize()));
```

Comment:
Isn't copy before removing safer?
```suggestion
        final ItemStack copy = stack.copy();
        if (helper.clear(copy)) {
            this.setCarried(copy.copyWithCount(stack.getMaxStackSize());
            return;
        }
```